c7n/resources/ami.py

```diff
@@ -74,3 +77,44 @@ class ImageAgeFilter(AgeFilter):
         'image-age',
         op={'type': 'string', 'enum': OPERATORS.keys()},
         days={'type': 'number', 'minimum': 0})
+
+
+@filters.register('unused')
+class ImageUnusedFilter(Filter):
+
+    schema = type_schema('unused', value={'type': 'boolean'})
+
+    def _pull_asg_images(self, results):
+        asg_manager = ASG(self.manager.ctx, {})
+        asgs = asg_manager.resources()
+        lcfgs = [a['LaunchConfigurationName'] for a in asgs
```

Comment:
this would be better for comparisons as a set comprehension.

also why only consider asgs with more than one instance?